test/test-clang-cl.sh

```diff
@@ -0,0 +1,40 @@
+#!/bin/bash
+#
+# Copyright (c) 2023 Huang Qinjin
+#
+# Permission to use, copy, modify, and/or distribute this software for any
+# purpose with or without fee is hereby granted, provided that the above
+# copyright notice and this permission notice appear in all copies.
+#
+# THE SOFTWARE IS PROVIDED "AS IS" AND THE AUTHOR DISCLAIMS ALL WARRANTIES
+# WITH REGARD TO THIS SOFTWARE INCLUDING ALL IMPLIED WARRANTIES OF
+# MERCHANTABILITY AND FITNESS. IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR
+# ANY SPECIAL, DIRECT, INDIRECT, OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+# WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN AN
+# ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION, ARISING OUT OF
+# OR IN CONNECTION WITH THE USE OR PERFORMANCE OF THIS SOFTWARE.
+
+. "${0%/*}/test.sh"
+
+cd "$TESTS"
+
+MSVC="${1:-/opt/msvc}"
+
+for arch in i686 x86_64 armv7 aarch64; do
+    case $arch in
+    i*86) msvcarch=x86 ;;
+    x86_64) msvcarch=x64 ;;
+    arm*) msvcarch=arm ;;
+    aarch64) msvcarch=arm64 ;;
+    esac
```

Comment:
I thinks this block can be moved to `msvcenv-native.sh` as it is also needed for users. And the logic should be reversed: construct clang triple from `ARCH` (sourced from `msvcenv.sh`).

Reply:
Oh, that's indeed a good idea. We could probably even use `$ARCH` right away as such from there, but could also make `msvcenv-native.sh` produce a full triple.

Anyway, that doesn't help for the cases when using `/winsysroot`, before sourcing `msvcenv-native.sh` though. But I guess we can reverse this and primarily use the MSVC arch names here, and then map this as necessary within `test-clang-cl-cmds.sh`.